protocol-designer/src/components/StepEditForm/fields/TipPositionField/ZTipPositionModal.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from 'react'
+import { createPortal } from 'react-dom'
+import cx from 'classnames'
+import { useTranslation } from 'react-i18next'
+import {
+  AlertModal,
+  Flex,
+  HandleKeypress,
+  Icon,
+  InputField,
+  OutlineButton,
+  RadioGroup,
+} from '@opentrons/components'
+import { getMainPagePortalEl } from '../../../portals/MainPageModalPortal'
+import { getIsTouchTipField } from '../../../../form-types'
+import { TipPositionZAxisViz } from './TipPositionZAxisViz'
+import * as utils from './utils'
+import { LARGE_STEP_MM, SMALL_STEP_MM, TOO_MANY_DECIMALS } from './constants'
+
+import type { StepFieldName } from '../../../../form-types'
+
+import modalStyles from '../../../modals/modal.module.css'
+import styles from './TipPositionInput.module.css'
+
+interface Props {
+  closeModal: () => void
+  isIndeterminate?: boolean
+  mmFromBottom: number | null
+  name: StepFieldName
+  updateValue: (val: number | null | undefined) => unknown
+  wellDepthMm: number
+}
+
+export const ZTipPositionModal = (props: Props): JSX.Element => {
+  const { isIndeterminate, name, wellDepthMm } = props
+  const { t } = useTranslation(['modal', 'button'])
+  const defaultMmFromBottom = utils.getDefaultMmFromBottom({
+    name,
+    wellDepthMm,
+  })
+
+  const [value, setValue] = React.useState<string | null>(
+    props.mmFromBottom === null ? null : String(props.mmFromBottom)
```

Comment:
related to the above comment
```suggestion
   mmFromBottom === null ? null : String(mmFromBottom)
```